src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -286,6 +287,9 @@ function setInitialDOMProperties(
       propKey === SUPPRESS_HYDRATION_WARNING
     ) {
       // Noop
+    } else if (propKey === AUTOFOCUS) {
+      // We polyfill it separately on the client during commit.
+      // We blacklist it here rather than in the property list because we emit it in SSR.
```

Comment:
Seems like it's time to fork that file then.

Reply:
I want to rework how it's done entirely (see https://github.com/facebook/react/pull/10805), and it would be more convenient if it was still centralized before I did that.

Reply:
I really don't like that we have this extra special case here. :/ Mostly because I hope we can simplify this whole thing and special cases might hang around.

We could instead special case it on the server to minimize code. But I'm not sure that's any better for the maintainability. At least with this it's clear that it is probably the client that needs to  change in the future.